cmake/docs/docs.py

```diff
@@ -73,7 +84,16 @@ def process_file(file_name, outdir):
     out_fn = os.path.join(outdir, out_fn)
     os.makedirs(os.path.dirname(out_fn), exist_ok=True)
     # Open both files, and loop over all the lines reading and writing each
-    print("[{}]({})".format(os.path.basename(out_fn).replace(".md", ""), relative_fn))
+    if Path(file_name).parent != CURRENT_DIR:
+        CURRENT_DIR = Path(file_name).parent
+        CURRENT_OFFSET = len(Path(file_name).relative_to(CURRENT_DIR).parents)
```

Comment:
Offset is probably not needed, I wanted to nest subfolders but couldn't get it to work easily.
This results in 
```
## toolchain
[...]
  - [toolchain-template](./toolchain/toolchain-template.md)
  - [arm-sf-linux](./toolchain/arm-sf-linux.md)
## toolchain/helpers
  - [arm-linux-base](./toolchain/helpers/arm-linux-base.md)
```